nexus/src/app/background/tasks/vpc_routes.rs

```diff
@@ -221,7 +229,33 @@ impl BackgroundTask for VpcRouteManager {
                     // number.
                     match &set.version {
                         Some(v) if !v.is_replaced_by(&version) => {
-                            continue;
+                            info!(
+                                log,
+                                "VPC route manager sled {} push not needed",
+                                sled.id()
+                            );
+                            //TODO(ry) continue;
+                            // Currently, this version is bumped in response to
+                            // events that change the routes. This has to be
+                            // done explicitly by the programmer in response to
+                            // any event that may result in a difference in
+                            // route resolution calculation. With the
+                            // introduction of internet gateway targets that
+                            // are parameterized on source IP, route resolution
+                            // computations can change based on events that are
+                            // not directly modifying VPC routes - like the
+                            // linkiage of an IP pool to a silo, or the
+                            // allocation of an external IP address and
+                            // attachment of that IP address to a service or
+                            // instance. This broadened context for changes
+                            // influencing route resolution makes manual
+                            // tracking of a router version easy to get wrong
+                            // and I feel like it will be a bug magnet.
+                            //
+                            // I think we should move decisions around change
+                            // propagation to be based on actual delta
+                            // calculation, rather than trying to manually
+                            // maintain a signal.
```

Comment:
Calling this comment out as something to discuss in this PR. CC @FelixMcFelix

Reply:
Yeah, I had maybe overlooked the scope of things that could change in connection with an Internet Gateway. That's a big surface area to keep an eye on... 

I'm not really sure how we would want to work around this, however. The motivation for using router versions was more for performance rather than correctness. My intent was to prevent each Nexus's copy of the background task from querying every Project's/VPC's/Subnet's state on a minute-by-minute (and per-wakeup) basis, since there are a lot of tables which are referenced during route resolution. This partly stems from the granularity of background task wakeups, which don't AFAIK allow a selective wakeup. So my initial feeling is that having Nexus store resolved tables and then send deltas is orthogonal to that?

It could also, in practice, not be too expensive to just rebuild VPC routing tables periodically! In which case, the places where we perform version bumps and/or background task wakeups would keep the system responsive rather than be crucial for correctness. If cost remains a concern, I wonder if we can just use `modified` timestamps in connected tables to drive the route update process (which would, I hope, be less error-prone).